mythril/laser/smt/bitvec.py

```diff
@@ -242,8 +580,16 @@ def Concat(*args: Union[BitVec, List[BitVec]]) -> BitVec:
 
     nraw = z3.Concat([a.raw for a in bvs])
     annotations = [] # type: Annotations
+    bitvecfunc = False
     for bv in bvs:
         annotations += bv.annotations
+        if isinstance(bv, BitVecFunc):
+            bitvecfunc = True
+
+    if bitvecfunc:
+        # Is there a better value to set name and input to in this case?
+        return BitVecFunc(raw=nraw, name=None, input=None, annotations=bv.annotations)
```

Comment:
I don't have a clear idea of this part but `annotations=bv.annotations` doesn't make when bv is being used after the termination of the for loop in line 584.

Reply:
You're right, good catch.